mlflow/pytorch.py

```diff
@@ -0,0 +1,133 @@
+    path = os.path.abspath(path)
+    if os.path.exists(path):
+        raise Exception("Path '{}' already exists".format(path))
+    os.makedirs(path)
+    model_path = os.path.join(path, "model.pth")
+
+    # Save pytorch model
+    torch.save(pytorch_model, model_path, **kwargs)
+    model_file = os.path.basename(model_path)
+
+    mlflow_model.add_flavor(FLAVOR_NAME, model_data=model_file, pytorch_version=torch.__version__)
+    pyfunc.add_to_model(mlflow_model, loader_module="mlflow.pytorch",
+                        data=model_file, env=conda_env)
+    mlflow_model.save(os.path.join(path, "MLmodel"))
+
+
+def _load_model(path, **kwargs):
+    mlflow_model_path = os.path.join(path, "MLmodel")
+    if not os.path.exists(mlflow_model_path):
+        raise RuntimeError("MLmodel is not found at '{}'".format(path))
+
+    mlflow_model = Model.load(mlflow_model_path)
+
+    assert FLAVOR_NAME in mlflow_model.flavors, \
+        "Stored model can not be loaded with mlflow.pytorch"
+
+    # This maybe replaced by a warning and then try/except torch.load
+    flavor = mlflow_model.flavors[FLAVOR_NAME]
+    assert torch.__version__ == flavor["pytorch_version"], \
+        "Unfortunately stored model version '{}' does not match "\
+        .format(flavor["pytorch_version"]) + \
+        "installed PyTorch version '{}'".format(torch.__version__)
+
+    path = os.path.abspath(path)
+    path = os.path.join(path, mlflow_model.flavors[FLAVOR_NAME]['model_data'])
+    return torch.load(path, **kwargs)
+
+
+def load_model(path, run_id=None, **kwargs):
+    """
+    Load a PyTorch model from a local file (if run_id is None) or a run.
+    :param path: Local filesystem path or Run-relative artifact path to the model saved by
+        `mlflow.pytorch.log_model`.
+    :param run_id: Run ID. If provided it is combined with path to identify the model.
+    :param kwargs: kwargs to pass to `torch.load` method
+    """
+    if run_id is not None:
+        path = mlflow.tracking._get_model_log_dir(model_name=path, run_id=run_id)
+
+    return _load_model(path, **kwargs)
+
+
+def load_pyfunc(path, **kwargs):
+    """
+    Load the model as PyFunc.
+    :param path: Local filesystem path to the model saved by `mlflow.pytorch.log_model`.
+    :param kwargs: kwargs to pass to `torch.load` method
+    """
+    return _PyTorchWrapper(_load_model(os.path.dirname(path), **kwargs))
+
+
+class _PyTorchWrapper(object):
+    """
+    Wrapper class that creates a predict function such that
+    predict(data: ndarray) -> model's output as numpy.ndarray
+    """
+    def __init__(self, pytorch_model):
+        self.pytorch_model = pytorch_model
+
+    def predict(self, data, device='cpu'):
+        assert isinstance(data, pd.DataFrame), "Input data should be pandas.DataFrame"
+        self.pytorch_model.eval()
+        with torch.no_grad():
+            input_tensor = torch.from_numpy(data.values.astype(np.float32)).to(device)
+            preds = self.pytorch_model(input_tensor)
```

Comment:
Q: do you know if `self.pytorch_model(input_tensor)` always outputs a single tensor? Was trying to look at the docs for `torch.nn.Module.__call__`, ([see here](https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html#Module)) the return type of `__call__` doesn't seem to be documented anywhere

Reply:
Looking at this [example](https://github.com/pytorch/examples/blob/master/word_language_model/main.py#L137), it seems that a model might be able to return multiple values. Maybe for now we can raise an exception in that case & think about how to support multi-output models in the future. Alternatively, we could try putting each output tensor into its own DataFrame column (although we'd need to be careful about documenting the ordering of the output columns). Either way, we should document the behavior in our RST docs

Reply:
Actually there is no restristriction on the number of tensors in the input and in the output of the model. This should be also true for other frameworks like TF and Keras.
However I think such check should be done on the initialization time and not at `prediction`. However, in PyTorch such introspection can be rather difficult...
I think it would be better to mention in the documentation that models used with pyfunc should be single input tensor of shape `(N, M)` and single output tensor of shape `(N, K)`.

Reply:
Got it, yeah mentioning the restriction on model input/output (one input tensor & one output tensor) in the documentation makes sense to me. IMO it would still be nice to raise an exception here if `preds` isn't a tensor, because otherwise (e.g. if `preds` is a tuple) the user will see an error like "tuple has no attribute `numpy()`", which doesn't give them an idea of how to fix the error. 

How about something like:
```
if not isinstance(preds, torch.Tensor):
    raise RuntimeError("Expected PyTorch model to output a single output tensor, but got output of type %s" % type(preds))
```